// rebass/dist/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = undefined;

var _systemComponents = require('system-components');

var _systemComponents2 = _interopRequireDefault(_systemComponents);

var _styledSystem = require('styled-system');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button = exports.Button = (0, _systemComponents2.default)({
  is: 'button',
  fontSize: 1,
  fontWeight: 'bold',
  lineHeight: 16 / 14,
  m: 0,
  px: 3,
  py: 2,
  color: 'white',
  bg: 'blue',
  borderRadius: 2,
  border: 0
}, function (props) {
  return {
    fontFamily: 'inherit',
    WebkitFontSmoothing: 'antialiased',
    display: 'inline-block',
    verticalAlign: 'middle',
    textAlign: 'center',
    textDecoration: 'none',
    appearance: 'none',
    '&:hover': {
      boxShadow: 'inset 0 0 0 999px ' + (0, _styledSystem.themeGet)('colors.darken.0')(props)
    },
    '&:focus': {
      outline: 0,
      boxShadow: '0 0 0 2px ' + (0, _styledSystem.themeGet)('colors.blue')(props)
    },
    '&:active': {
      backgroundColor: (0, _styledSystem.themeGet)('colors.blue.6')(props),
      boxShadow: 'inset 0 0 8px ' + (0, _styledSystem.themeGet)('colors.darken.1')(props)
    },
    '&:disabled': {
      opacity: 1 / 4
    }
  };
});

Button.displayName = 'Button';

exports.default = Button;
