// rebass/dist/utils.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.carouselIndex = exports.drawerOpen = exports.left = exports.bottom = exports.right = exports.top = exports.zIndex = exports.sizeHeight = exports.sizeWidth = exports.minHeight = exports.maxHeight = exports.maxWidth = exports.backgroundPosition = exports.backgroundSize = exports.backgroundImage = exports.ratio = exports.fontFamily = exports.darken = undefined;

var _styledSystem = require('styled-system');

var _styledSystem2 = _interopRequireDefault(_styledSystem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var darken = exports.darken = function darken(n) {
  return 'rgba(0, 0, 0, ' + n + ')';
};

var fontFamily = exports.fontFamily = _styledSystem2.default.style({
  prop: 'font',
  key: 'fonts',
  cssProperty: 'fontFamily'
});

var ratio = exports.ratio = function ratio(props) {
  return props.ratio ? {
    height: 0,
    paddingBottom: props.ratio * 100 + '%'
  } : null;
};

var backgroundImage = exports.backgroundImage = function backgroundImage(props) {
  return props.src ? {
    backgroundImage: 'url(' + props.src + ')'
  } : null;
};

var backgroundSize = exports.backgroundSize = _styledSystem2.default.style({
  prop: 'backgroundSize'
});

var backgroundPosition = exports.backgroundPosition = _styledSystem2.default.style({
  prop: 'backgroundPosition'
});

var maxWidth = exports.maxWidth = _styledSystem2.default.responsiveStyle({
  prop: 'maxWidth',
  cssProperty: 'maxWidth',
  key: 'maxWidths',
  numberToPx: true
});

var maxHeight = exports.maxHeight = _styledSystem2.default.responsiveStyle({
  prop: 'maxHeight',
  cssProperty: 'maxHeight',
  key: 'maxHeights',
  numberToPx: true
});

var minHeight = exports.minHeight = _styledSystem2.default.responsiveStyle({
  prop: 'minHeight',
  cssProperty: 'minHeight',
  numberToPx: true
});

var sizeWidth = exports.sizeWidth = _styledSystem2.default.responsiveStyle({
  prop: 'size',
  cssProperty: 'width',
  key: 'sizes',
  numberToPx: true
});

var sizeHeight = exports.sizeHeight = _styledSystem2.default.responsiveStyle({
  prop: 'size',
  cssProperty: 'height',
  key: 'sizes',
  numberToPx: true
});

var zIndex = exports.zIndex = _styledSystem2.default.responsiveStyle({
  prop: 'zIndex',
  cssProperty: 'zIndex'
});

var top = exports.top = _styledSystem2.default.responsiveStyle({
  prop: 'top',
  cssProperty: 'top',
  numberToPx: true
});

var right = exports.right = _styledSystem2.default.responsiveStyle({
  prop: 'right',
  cssProperty: 'right',
  numberToPx: true
});

var bottom = exports.bottom = _styledSystem2.default.responsiveStyle({
  prop: 'bottom',
  cssProperty: 'bottom',
  numberToPx: true
});

var left = exports.left = _styledSystem2.default.responsiveStyle({
  prop: 'left',
  cssProperty: 'left',
  numberToPx: true
});

var drawerOpen = exports.drawerOpen = function drawerOpen(_ref) {
  var open = _ref.open,
      position = _ref.position,
      size = _ref.size;

  var h = /^(left|right)$/.test(position) ? 1 : 0;
  var width = h ? size + 'px' : null;
  var height = h ? null : size + 'px';
  var transforms = {
    left: 'translateX(-100%)',
    right: 'translateX(100%)',
    top: 'translateY(-100%)',
    bottom: 'translateY(100%)'
  };
  var top = /^(top|left|right)$/.test(position) ? 0 : null;
  var bottom = /^(bottom|left|right)$/.test(position) ? 0 : null;
  var left = /^(left|top|bottom)$/.test(position) ? 0 : null;
  var right = /^(right|top|bottom)$/.test(position) ? 0 : null;

  return {
    width: width,
    height: height,
    top: top,
    bottom: bottom,
    left: left,
    right: right,
    transform: open ? null : transforms[position]
  };
};

var carouselIndex = exports.carouselIndex = function carouselIndex(_ref2) {
  var index = _ref2.index;
  return {
    '& > div:first-child': {
      marginLeft: index * 100 + '%',
      transitionProperty: 'margin',
      transitionDuration: '.2s',
      transitionTimingFunction: 'ease-out'
    }
  };
};
